### DISCUSSION TITLE
Twikoo doesn't work until the page is refreshed.

### DISCUSSION THREAD

**Discussion by ghost** [HEIGE-PCloud/DoIt#488] [Q&A] [answered]
### Describe the bug 描述你遇到的错误

打开我的任意一篇[文章](https://www.bahuangshanren.tech/2022-3/)，发现twikoo并不显示，刷新网页面之后，twikoo才出现。本地预览时正常，只有部署后的网页才有这个问题。

### Build Environment 构建环境

- OS: Windows 10
- Theme version/commit: https://github.com/HEIGE-PCloud/DoIt/commit/5f0ec9683d76647303f8944181a0f5befc73a9b7
- Hugo version: 0.92.2, extended

### Preview Environment 预览环境

- OS: Windows 10
- Browser: Microsoft Edge
- Hugo version: 0.92.2, extended

### Additional Information 补充信息

为了尽量减少影响，我目前通过子模块的方式引用主题，本地构建后使用[peaceiris/actions-gh-pages@v3.7.3](https://github.com/peaceiris/actions-gh-pages)部署[public](https://github.com/BahuangShanren/blog/tree/master/public)目录。

我试图通过控制变量进行排查，在以下若干次尝试中，**问题仍然存在**：

- 换用GitHub Actions或者Cloudflare Pages来构建。
- twikoo部署到腾讯云或者vercel。
- 取消Cloudflare的代理。
- 直接构建部署DoIt的[Example Site](https://github.com/HEIGE-PCloud/DoIt/tree/main/exampleSite)。
- 参考 #469 ，我在本地`themes/DoIt`目录下运行`npm install`。

我的[配置文件](https://github.com/BahuangShanren/blog/blob/master/config.toml)

**Comment by HEIGE-PCloud** [comment, marked as answer]
<img width="564" alt="image" src="https://user-images.githubusercontent.com/52968553/155903789-bcd7d5da-a034-441b-b28c-68e12ac31a4e.png">

我注意到所有 pjax-asset 的 type 在不同页面都被做出了不同的修改。可能是因此造成的 Pjax 加载问题。我在本地和 exampleSite 中均未能复现出这个问题。请问在部署环境中还做过什么其他配置吗？

**Reply by ghost** [reply]
@HEIGE-PCloud 为了减少影响因素，我没有进行其他配置。这是我的[GitHub Actions](https://github.com/BahuangShanren/blog/blob/master/.github/workflows/gh-pages.yml)文件。
由于我不是很确定DoIt的构建部署流程，所以与您做过的尝试可能并不一样。DoIt是用[/.github/workflows/Build.yml](https://github.com/HEIGE-PCloud/DoIt/blob/main/.github/workflows/Build.yml)构建，然后通过Cloudflare Pages直接发布（除发布外不执行其他操作）吗？
我尝试过模仿这个过程，但问题仍然出现。

另一个猜想，问题可能与GitHub Actions文件中的`npm install`有关？
我没有成功地完成这一步，因为GitHub Actions会将node_modules安装在项目根目录，而DoIt的examplesite是向上层路径引用主题的，所以能正确使用node_modules，但如果主题使用子模块方式引入，那么DoIt主题并不能使用到node_modules的文件。我尝试过`npm install --prefix`指定安装路径，但只安装了部分依赖。

如果您能提供一个示例仓库，主题使用子模块方式，并以一个简单的GitHub Actions来构建部署，那再好不过了。

**Reply by HEIGE-PCloud** [reply]
https://heige-pcloud.github.io/test-blog/

我尝试部署了你的博客 似乎没有遇到任何问题

https://github.com/HEIGE-PCloud/test-blog/

**Reply by ghost** [reply]
感谢您的热心帮助❤，我已经解决了问题✅。

我之前提出的解决方式是错误的，估计是缓存方面的影响让我误以为解决了问题，结果后来这个问题再次出现😨。

我已经重新编辑了此回答，并保证这是正确的解决方法：

Cloudflare站点→速度→优化→关闭`Rocket Loader`

![image](https://user-images.githubusercontent.com/48022216/156865976-8245ce64-7312-439d-b08c-a0fc54d6ba27.png)
